ammico/video_summary.py

```diff
@@ -0,0 +1,1136 @@
+                        )
+                else:
+                    generated_ids = self.summary_model.model.generate(
+                        **processor_inputs, generation_config=gen_conf
+                    )
+            except RuntimeError as e:
+                warnings.warn(
+                    f"Generation failed with error: {e}. Retrying with cuDNN disabled.",
+                    RuntimeWarning,
+                )
+                cudnn_was_enabled = (
+                    torch.backends.cudnn.is_available() and torch.backends.cudnn.enabled
+                )
+                if cudnn_was_enabled:
+                    torch.backends.cudnn.enabled = False
+                try:
+                    generated_ids = self.summary_model.model.generate(
+                        **processor_inputs, generation_config=gen_conf
+                    )
+                except Exception as retry_error:
+                    raise RuntimeError(
+                        f"Failed to generate ids after retry: {retry_error}"
+                    ) from retry_error
+                finally:
+                    if cudnn_was_enabled:
+                        torch.backends.cudnn.enabled = True
+
+        decoded = self._decode_trimmed_outputs(
+            generated_ids, processor_inputs, tokenizer, prompt_texts
+        )
+        return decoded
+
+    def _audio_to_text(self, audio_path: str) -> List[Dict[str, Any]]:
+        """
+        Convert audio file to text using an whisper model.
+        Args:
+            audio_path (str): Path to the audio file.
+        Returns:
+            List[Dict[str, Any]]: List of transcribed audio segments with start_time, end_time, text, and duration.
+        """
+
+        if not os.path.exists(audio_path):
+            raise ValueError(f"Audio file {audio_path} does not exist.")
+
+        try:
+            torch.backends.cuda.matmul.allow_tf32 = True
+            torch.backends.cudnn.allow_tf32 = True
+            audio = whisperx.load_audio(audio_path)
+            transcribe_result = self.audio_model.model.transcribe(audio)
+            model_a, metadata = whisperx.load_align_model(
+                language_code=transcribe_result["language"],
+                device=self.audio_model.device,
+            )
+            aligned_result = whisperx.align(
+                transcribe_result["segments"],
+                model_a,
+                metadata,
+                audio,
+                self.audio_model.device,
+            )
+            audio_descriptions = []
+            for segment in aligned_result["segments"]:
+                audio_descriptions.append(
+                    {
+                        "start_time": segment["start"],
+                        "end_time": segment["end"],
+                        "text": segment["text"].strip(),
+                        "duration": segment["end"] - segment["start"],
+                    }
+                )
+            return audio_descriptions
```

Comment:
Can we add the audio descriptions to the `video_dict` too? I think it would be good at least for debugging, but maybe Delia also wants to do a further topic analysis on these.